src/NuGet.Core/NuGet.PackageManagement/NuGetPackageManager.cs

```diff
@@ -135,12 +132,39 @@ public class NuGetPackageManager
 
             InitializePackagesFolderInfo(PackagesFolderPathUtility.GetPackagesFolderPath(SolutionManager, Settings), excludeVersion);
             DeleteOnRestartManager = deleteOnRestartManager;
+        }
 
-            string globalPackagesFolder = BuildIntegratedProjectUtility.GetEffectiveGlobalPackagesFolderOrNull(solutionManager.SolutionDirectory, settings);
-            if (globalPackagesFolder != null)
+        /// <summary>
+        /// SourceRepositories for the user global package folder and all fallback package folders.
+        /// </summary>
+        public IReadOnlyList<SourceRepository> GlobalPackageFolderRepositories
+        {
+            get
             {
-                var globalPackagesSource = new Configuration.PackageSource(globalPackagesFolder);
-                GlobalPackagesFolderSourceRepository = SourceRepositoryProvider.CreateRepository(globalPackagesSource, FeedType.FileSystemV3);
+                if (_globalPackageFolderRepositories == null)
+                {
+                    var sources = new List<SourceRepository>();
```

Comment:
that could cause a race case, the first call could be in the middle of creating the repos while the second gets an empty list.